src/stactools/goes_glm/stac.py

```diff
@@ -174,130 +175,168 @@ def create_item(
         collection (pystac.Collection): HREF to an existing collection
         nogeoparquet (bool): If set to True, no geoparquet file is generated for the Item
         nonetcdf (bool): If set to True, the netCDF file is not added to the Item
+        fixnetcdf (bool): If set to True, fixes missing _Unsigned attributes in some of
+            the older netCDF files
 
     Returns:
         Item: STAC Item object
     """
 
-    dataset = Dataset(asset_href, "r", format="NETCDF4")
-
-    id = dataset.dataset_name.replace(".nc", "")
-    sys_env = id[:2]
-    if sys_env != "OR":
-        logger.warning("You are ingesting test data.")
-
-    computed_datetime = center_datetime(
-        dataset.time_coverage_start, dataset.time_coverage_end
-    )
-
-    try:
-        platform = constants.Platforms[dataset.platform_ID]
-        if platform == constants.Platforms.G18:
-            raise Exception("GOES-18/T is not supported yet")
-    except ValueError:
-        raise Exception(
-            f"The dataset contains an invalid platform identifier: {dataset.platform_ID}"
-        )
+    with Dataset(asset_href, "a", format="NETCDF4") as dataset:
```

Comment:
Always opening the files for reading is not ideal. On the other hand, scaling the variables manually also doesn't seem ideal.